app/controllers/facility_accounts_controller.rb

```diff
@@ -70,7 +47,32 @@ def show
   # GET /facilities/:facility_id/accounts/new
   def new
     @owner_user = User.find(params[:owner_user_id])
-    @account    = @owner_user.accounts.new(:expires_at => Time.zone.now + 1.year)
+    @available_account_types = available_account_types
+    @current_account_type = current_account_type
+    @account = current_account_type.constantize.new
+  end
+
+  # POST /facilities/:facility_id/accounts
+  def create
+    @owner_user = User.find(params[:owner_user_id])
+    @available_account_types = available_account_types
+    @current_account_type = current_account_type
+
+    @account = AccountBuilder.for(params[:account_type]).new({
+      account_type: params[:account_type],
```

Comment:
If you are getting the accountbuilder with the factory method, is it possible to avoid also passing it in via the constructor?